moabb/evaluations/splitters.py

```diff
@@ -101,3 +109,112 @@ def split(self, y, metadata):
                 for train_ix, test_ix in splitter.split(indices, y_session):
 
                     yield indices[train_ix], indices[test_ix]
+
+
+class CrossSessionSplitter(BaseCrossValidator):
+    """Data splitter for cross session evaluation.
+
+    Cross-session evaluation uses a Leave-One-Session-Out cross-validation to
+    evaluate performance across sessions, but for a single subject. This splitter
+    assumes that all data from all subjects is already known and loaded.
+
+    .. image:: images/crosssess.jpg
+        :alt: The schematic diagram of the CrossSession split
+        :align: center
+
+    The inner cross-validation strategy can be changed by passing the
+    `cv_class` and `cv_kwargs` arguments. By default, it uses LeaveOneGroupOut,
+    which effectively performs Leave-One-Session-Out cross-validation when
+    sessions are passed as groups.
+
+    Parameters
+    ----------
+    shuffle : bool, default=False
+        Whether to shuffle the session order for each subject.
+    random_state: int, RandomState instance or None, default=None
+        Controls the randomness when `shuffle` is True.
+        Pass an int for reproducible output across multiple function calls.
+    cv_class: cross-validation class, default=LeaveOneGroupOut
+        Inner cross-validation strategy for splitting the sessions.
+        For cross-session splitting, LeaveOneGroupOut is the most suitable as default.
+    cv_kwargs: dict
+        Additional arguments to pass to the inner cross-validation strategy.
+    """
+
+    def __init__(
+        self,
+        shuffle: bool = False,
+        random_state: int = None,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        **cv_kwargs: dict,
```

Comment:
What do you think about this instead:
```suggestion
        cv_kwargs: dict[str, Any] | None = None,
```

If you prefer to keep the current version, there is no type hint for `**kwargs` (https://peps.python.org/pep-0692/):
```suggestion
        **cv_kwargs,
```

Reply:
I prefer your suggestion, but I was trying to keep compatible with the tom and Bruna WhitinSession Splitter. On one hand it's less verbose, on the other it's kind of bad for typing. Any input here @tomoreau?

Reply:
We could also update `WhitinSessionSplitter`'s parameters. It's still an experimental class

Reply:
I am removing the type hint until @tomMoral gives the input.... let's merging following his original design to avoid future probs